### PR TITLE
[DataObjects] Layout Component - Dynamic Text refactoring

### LINKED ISSUE #5599: Containerize Text Renderers
https://github.com/pimcore/pimcore/blob/master/models/DataObject/ClassDefinition/Layout/Text.php#L135

### PR DESCRIPTION
**Pull request by dvesh3**
## Changes in this pull request  
Resolves #5599

### CODE REVIEW

File: models/DataObject/ClassDefinition/Layout/Text.php [hunk 1]
```diff
@@ -126,15 +126,15 @@ public function setBorder(bool $border): void
...
-                $context['fieldname'] = $this->getName();
-                $context['layout'] = $this;
-                $result = call_user_func($renderingClass . '::renderLayoutText', $this->renderingData, $object, $context);
-                $this->html = $result;
-            }
+        $renderingClass = Model\DataObject\ClassDefinition\Helper\DynamicTextResolver::resolveRenderingClass(
+            $this->getRenderingClass()
+        );
+
+        if (method_exists($renderingClass, 'renderLayoutText')) {
```
Comment (weisswurstkanone) [inline models/DataObject/ClassDefinition/Layout/Text.php]: maybe instanceof DynamicTextLabelInterface in addition (keep method_exists for BC reasons).

File: models/DataObject/Service.php [hunk 2]
```diff
@@ -946,7 +946,10 @@ public static function getValidLayouts(Concrete $object)
         }
 
         foreach ($list as $customLayout) {
-            $resultList[$customLayout->getId()] = $customLayout;
+            if($customLayout instanceof ClassDefinition\CustomLayout) {
```
Comment (dpfaffenbauer) [inline models/DataObject/Service.php]: doesn't belong to this PR, does it?

File: models/DataObject/ClassDefinition/Layout/Text.php (hunk 1 above)
Comment (brusch) [reply models/DataObject/ClassDefinition/Layout/Text.php]: Good point, I'll add it.

**Comment by dpfaffenbauer** [comment]
did you test if that is also backward-compatible? Cause the Class-Resolver makes a "new $class" if it doesn't come from DI. AFAIK currently the layout used a static function

File: models/DataObject/Service.php (hunk 2 above)
Comment (dvesh3) [reply models/DataObject/Service.php]: you right this does not belong to this PR....my bad! it was miss :/

Comment (brusch) [reply models/DataObject/Service.php]: But doesn't hurt, so i kept it.

**Comment by dvesh3** [comment]
@dpfaffenbauer yes it backward compatible... please see https://github.com/pimcore/pimcore/pull/5696/files#diff-f60d77aacb93b63df6601aa3bb681637R136

**Comment by dpfaffenbauer** [comment]
@dvesh3 right, just checked `call_user_func` works with static functions on instances without notices. (https://www.php.net/manual/en/function.call-user-func.php#example-6382)

**Comment by brusch** [comment]
Added a deprecation warning and fixed the example in the docs.
